### PR TITLE
Migrate all inference api call to client side

### PR DESCRIPTION
**Pull request by notmd**
Migrate all api calls of inference to the client side (no proxy through NodeJS anymore). I haven't verified the full UX of chat yet. Will do when I update the chat layout.

### CODE REVIEW

File: website/src/types/Chat.ts [hunk 1]
```diff
@@ -12,10 +12,10 @@ export interface ChatItem {
   id: string;
   created_at: string; //timestamp
   modified_at: string; //timestamp
+  messages: InferenceMessage[];
```
Comment (notmd) [inline website/src/types/Chat.ts:15]: Inference backend always returns an array even when there is no message yet.

File: website/src/hooks/chat/useInferenceAuth.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const useInferenceAuth = () => {
+  const { data, isLoading, mutate } = useSWR<boolean>("/auth/check", () => {
+    return new OasstInferenceClient().check_auth();
+  });
+
+  return {
+    isAuthenticated: data,
+    isLoading,
+    mutate,
```
Comment (AbdBarho) [inline website/src/hooks/chat/useInferenceAuth.ts:12]: could you please rename to something that reflects the function? like `refreshAuth` or similar?

File: website/src/pages/chat/[id].tsx [hunk 3]
```diff
@@ -45,19 +44,14 @@ const Chat = ({ id, modelInfos }: ChatProps) => {
...
+export const getServerSideProps: GetServerSideProps<ChatProps, { id: string }> = async ({ locale = "en", params }) => {
   if (!isChatEnabled()) {
     return {
       notFound: true,
     };
   }
 
-  const token = await getToken({ req });
-  const client = createInferenceClient(token);
+  const client = new OasstInferenceClient();
```
Comment (AbdBarho) [inline website/src/pages/chat/[id].tsx:54]: Since we do all everything on the client, we should probably move this to the client at some point

**Review by AbdBarho — COMMENTED** [review]
I will have to look over this again

File: website/src/pages/chat/[id].tsx (hunk 3 above)
Comment (notmd) [reply website/src/pages/chat/[id].tsx:54]: For API the doesn't require auth, I want to do it in server side. It must more performance and the logic on the client less complicated (loading/error state).

**Pull request closed without merging**